cmd/contour/serve.go

```diff
@@ -213,6 +218,29 @@ func (ctx *serveContext) tlsconfig() *tls.Config {
 	}
 }
 
+// verifytlsflags indicates if the TLS flags are set up correctly.
```

Comment:
I think this function is doing more than one thing, it should probably do less and the things that it stops doing move elsewhere. Specifically the conflation of handling "tls flags are wrong, but its ok because we said --insecure". What I'd recommend is 

- a function that validates that the TLS parameters are correct, returning an error if not
- a function that decides to validate the tls parameters or not, which is probably using `DisableCertificateAuth` above.

after that point the logic to serve TLS or not is governed by `DisableCertificateAuth`. If `DisableCertificateAuth` is true, then we don't consult the TLS flags, if it is false then we use the TLS flags as they're validated correct above.

Reply:
That makes much more sense, thanks.